addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5862,10 +5862,12 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
         if (!this.originalSrc || !this._isImageSupportedForProcessing(img)) {
             return;
         }
-        const $select = $(uiFragment).find('we-select[data-name=format_select_opt]');
-        (await this._computeAvailableFormats()).forEach(([value, [label, targetFormat]]) => {
-            $select.append(`<we-button data-select-format="${Math.round(value)} ${targetFormat}" class="o_we_badge_at_end">${label} <span class="badge rounded-pill text-bg-dark">${targetFormat.split('/')[1]}</span></we-button>`);
-        });
+        if (this.mimetypeBeforeConversion) {
+            const $select = $(uiFragment).find('we-select[data-name=format_select_opt]');
+            (await this._computeAvailableFormats()).forEach(([value, [label, targetFormat]]) => {
```

Comment:
I feel that you leave the `_computeAvailableFormats` function buggy on its own if you only do that: could not you rather check `mimetypeBeforeConversion` in there and in case it is not set, return an empty list of format? In which case you would not have to change this code here?